py/server/deephaven/table_listener.py

```diff
@@ -87,48 +185,74 @@ def _do_locked(f: Callable, lock_type="shared") -> None:
         raise ValueError(f"Unsupported lock type: lock_type={lock_type}")
 
 
-def _nargs_listener(listener) -> int:
-    """Returns the number of arguments the listener takes.
+class TableListener(ABC):
+    """An abstract table listener class that should be subclassed by any user Table listener class."""
 
-    Args:
-        listener: listener
+    @abstractmethod
+    def onUpdate(self, *args, **kwargs) -> None:
```

Comment:
Since we've thrown out most of the legacy listener types, I think the signature of the `onUpdate` method can probably be tightened up as well to something like:
```
def on_update(self, is_replay: bool, update: WhateverTheType) -> None: